src/main/java/io/jenkins/plugins/coverage/CoverageProcessor.java

```diff
@@ -236,29 +229,36 @@ private void setDiffInCoverageForChangeRequest(CoverageResult coverageReport) {
             listener.getLogger().println("Build not found on target branch job, won't calculate coverage diff");
             return;
         }
+
         CoverageAction targetBranchCoverageAction = buildToTakeCoverageFrom.getAction(CoverageAction.class);
         if (targetBranchCoverageAction == null) {
             listener.getLogger().println("Coverage action not found on target branch build, won't calculate coverage diff");
             return;
         }
+
         CoverageResult targetBranchCoverageResult = targetBranchCoverageAction.getResult();
         if (targetBranchCoverageResult == null) {
             listener.getLogger().println("Coverage result not found on target branch coverage action, won't calculate coverage diff");
             return;
         }
-        Ratio targetBranchLinesCoverage = targetBranchCoverageResult.getCoverage(CoverageElement.LINE);
-        if (targetBranchLinesCoverage == null) {
-            listener.getLogger().println("Lines coverage not found on target branch, won't calculate coverage diff");
-            return;
-        }
-        float percentageDiff =
-                changeRequestLinesCoverage.getPercentageFloat() - targetBranchLinesCoverage.getPercentageFloat();
-        coverageReport.setChangeRequestCoverageDiffWithTargetBranch(percentageDiff);
-        coverageReport.setLinkToBuildThatWasUsedForComparison(buildToTakeCoverageFrom.getUrl());
+
+        Map<CoverageElement, Float> deltaCoverage = new TreeMap<>();
+        targetBranchCoverageResult.getResults().forEach((coverageElement, referenceRatio) -> {
+            Ratio buildRatio = coverageReport.getCoverage(coverageElement);
+
+            if (buildRatio != null) {
+                float diff = buildRatio.getPercentageFloat() - referenceRatio.getPercentageFloat();
```

Comment:
See comment below.

Reply:
Which comment below?

Reply:
Ah, actually above, that `Ratio` vs. `float` discussion.